src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/JSONCodecRegistry.java

```diff
@@ -156,27 +156,76 @@ public static <JavaT, CqlT> JSONCodec<JavaT, CqlT> codecToJSON(DataType targetCQ
           JSONCodec.ToCQL.unsafeIdentity(),
           JSONCodec.ToJSON.unsafeNodeFactory(JsonNodeFactory.instance::numberNode));
 
-  public static final JSONCodec<BigDecimal, BigDecimal> DECIMAL =
+  public static final JSONCodec<BigDecimal, BigDecimal> DECIMAL_FROM_BIG_DECIMAL =
       new JSONCodec<>(
           GenericType.BIG_DECIMAL,
           DataTypes.DECIMAL,
           JSONCodec.ToCQL.unsafeIdentity(),
           JSONCodec.ToJSON.unsafeNodeFactory(JsonNodeFactory.instance::numberNode));
 
-  public static final JSONCodec<BigDecimal, Double> DOUBLE =
+  private static final JSONCodec<BigInteger, BigDecimal> DECIMAL_FROM_BIG_INTEGER =
+      new JSONCodec<>(
+          GenericType.BIG_INTEGER,
+          DataTypes.DECIMAL,
+          // This is safe transformation, cannot fail:
+          (cqlType, value) -> new BigDecimal(value),
+          JSONCodec.ToJSON.unsafeNodeFactory(JsonNodeFactory.instance::numberNode));
+
+  private static final JSONCodec<Long, BigDecimal> DECIMAL_FROM_LONG =
+      new JSONCodec<>(
+          GenericType.LONG,
+          DataTypes.DECIMAL,
+          // This is safe transformation, cannot fail:
+          (cqlType, value) -> new BigDecimal(value),
+          JSONCodec.ToJSON.unsafeNodeFactory(JsonNodeFactory.instance::numberNode));
+
+  public static final JSONCodec<BigDecimal, Double> DOUBLE_FROM_BIG_DECIMAL =
       new JSONCodec<>(
           GenericType.BIG_DECIMAL,
           DataTypes.DOUBLE,
+          // TODO: bounds checks (over/underflow)
           JSONCodec.ToCQL.safeNumber(BigDecimal::doubleValue),
           JSONCodec.ToJSON.unsafeNodeFactory(JsonNodeFactory.instance::numberNode));
 
-  public static final JSONCodec<BigDecimal, Float> FLOAT =
+  private static final JSONCodec<BigInteger, Double> DOUBLE_FROM_BIG_INTEGER =
+      new JSONCodec<>(
+          GenericType.BIG_INTEGER,
+          DataTypes.DOUBLE,
+          // TODO: bounds checks (over/underflow)
+          JSONCodec.ToCQL.safeNumber(BigInteger::doubleValue),
+          JSONCodec.ToJSON.unsafeNodeFactory(JsonNodeFactory.instance::numberNode));
+
+  private static final JSONCodec<Long, Double> DOUBLE_FROM_LONG =
+      new JSONCodec<>(
+          GenericType.LONG,
+          DataTypes.DOUBLE,
+          JSONCodec.ToCQL.safeNumber(Long::doubleValue),
+          JSONCodec.ToJSON.unsafeNodeFactory(JsonNodeFactory.instance::numberNode));
+
+  public static final JSONCodec<BigDecimal, Float> FLOAT_FROM_BIG_DECIMAL =
       new JSONCodec<>(
           GenericType.BIG_DECIMAL,
           DataTypes.FLOAT,
+          // TODO: bounds checks (over/underflow)
```

Comment:
in my testing the under / overflow turns into a +/- Infinity

Reply:
@amorton Ok: that seems reasonable -- but can discuss if needs different handling